docs/articles/concepts/kafka.md

```diff
@@ -0,0 +1,167 @@
+to a higher value (eg: 1000). The `batch_size` parameter configures the maximum number of messages
+that will be fetched at a time from each worker.
+
+## Message Types
+
+Messages received from `KafkaSource` are of type [`KafkaSourceMessage`](
+/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSourceMessage).
+This dataclass includes basic Kafka fields like `.key` and `.value`, as well as
+extra information fields like `.headers`.
+
+Messages that are published to a `KafkaSink` must be of type
+[`KafkaSinkMessage`](/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSinkMessage).
+With the `.key` and `.value` fields set, or of type `KafkaSourceMessage`, if the message is unmodified.
+
+## Kafka and Recovery
+
+Typical deployments of Kafka utilize [consumer groups](https://developer.confluent.io/courses/architecture/consumer-group-protocol/)
+in order to manage partition assignment and the storing of Kafka offsets.
+
+Bytewax does **not** use consumer groups to store offsets or asssign partitions to consumers. In order
+to correctly support [recovery](/docs/concepts/recovery), Bytewax must manage and store the
+consumer offsets in recovery partitions.
+
+When recovery is not enabled, Bytewax will start consuming from each partition using the earliest
+available offset. This setting can be changed when creating a new `KafkaSource` with the
+`starting_offset` parameter, which accepts the types defined in the
+[`confluent_kafka`](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#offset)
+library.
+
+If you are not using recovery and would prefer to track offsets on the broker side, you can
+pass additional options to the Kafka input sources to create a consumer group:
+
+```python
+from confluent_kafka import OFFSET_STORED
+
+add_config = {"group.id": "consumer_group", "enable.auto.commit": "true"}
+BROKERS = ["localhost:19092"]
+IN_TOPICS = ["in_topic"]
+
+flow = Dataflow("kafka_in_out")
+kinp = kop.input(
+    "inp",
+    flow,
+    starting_offset=OFFSET_STORED,
+    add_config=add_config,
+    brokers=BROKERS,
+    topics=IN_TOPICS,
+)
+```
+
+## Partitions
+
+Partitions are a fundamental concept for Kafka and Redpanda, and are the unit of parallelism for
+producing and consuming messages.
+
+When multiple workers are started, Bytewax will internally assign individual partitions to the
+available number of workers. If there are fewer partitions than workers, some workers will not
+be assigned a partition to read from.
+
+If the number of partitions changes, Dataflows will need to be restarted in order to rebalance
+new partition assignments to workers.
+
+## Schema registry
+
+Bytewax supports integrating with the [Redpanda Schema Registry](https://docs.redpanda.com/current/manage/schema-reg/)
+as well as the [Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html).
```

Comment:
This lacks one more sentence about why would you want to do this at all, either

> Sometimes you wake up and wonder what if you want to be sure about the format of my messages, now you're in luck because schema registry does just that!

just kidding, but just some reasoning as to why that thing might be useful would be good to have here